src/BinaryTree/BinaryTree.cs

```diff
@@ -7,7 +7,7 @@ namespace BinaryTree
     public class BinaryTree<T> : ICollection<T> where T : IComparable<T>
     {
         private ITraversalStrategy<T> _traversalStrategy;
-        private BinaryTreeNode<T> _head;
+        public BinaryTreeNode<T> _head;
```

Comment:
You can't just simply make a public field. It is bad practice to use public fields in C# (see [here](https://softwareengineering.stackexchange.com/questions/161303/is-it-bad-practice-to-use-public-fields) and [here](https://stackoverflow.com/questions/1180860/public-fields-versus-automatic-properties)).